stock_picking_import_serial_number/wizard/import_serial_number_wizard.py

```diff
@@ -67,14 +71,39 @@ def action_import(self):
                 self._import_serial_number(xl_sheet, move_lines, picking)
         self.data_file = False
 
+    def _search_or_create_package(self, name):
+        """Return the package (if it exists) or create a new one."""
+        package_model = self.env["stock.quant.package"]
+        package = package_model.search(
+            [("name", "=", name), ("company_id", "=", self.env.company.id)], limit=1
+        )
+        return package or package_model.create({"name": name})
```

Comment:
Missing `company_id`

Reply:
`company_id` field is a compute store.

I think it should be taken into account for search but should be omitted in create because it will be automatically set with the quants information (https://github.com/odoo/odoo/blob/15.0/addons/stock/models/stock_quant .py#L1047)

Reply:
Uhm, then maybe we should ignore it also in search or sophisticate the search, as if the package is empty, it won't match incorrectly.

Reply:
Ok, ignored the field in search.

Reply:
I think the search should be `"|", ("company_id", "=", False), ("company_id", "=", self.picking_ids.company_id.id)`